### DISCUSSION TITLE
String s too long on force:mdapi:deploy and large number of test class to run

### DISCUSSION THREAD

**Discussion by Rihan9** [forcedotcom/cli#2260] [Ideas]
**Is your feature request related to a problem? Please describe.**
Hi,
I need to run a deploy throw sfdx with selective tests to run, but the cmd limit of 8k character block me.
Is possible to add something like #360 also for this?

**Describe the solution you'd like**
Add the list of test classes into a file and reference it with -f parameters, or something like this.

**Describe alternatives you've considered**
I try cygwin and powershell which have a character limit much higher without success. I hypothesize that sfdx spawns another cmd inside itself.

[a comment by an automated account is not shown]

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-9334120](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000N5jQYAS/view)

**Comment by amarek-cytiva** [comment]
Dear Support team,
do we know when this item goona be picked up?
With the previous ANT tool, there was no such limitation, in sfdx, it can potentially block the deployment when tiring to use deployment/validation with a big amount of unit tests specified.

**Comment by Rihan9** [comment]
@amarek-craftware yeah, I agree. A run all seems to be the only workaround, but it cost us around 2 to 4 more hour to validate.

**Comment by mshanemc** [comment]
The metadata API itself has a limit of how big the request can be, which is the next one you'd hit if you found a way around the terminal limit.

What you really want is the mdapi to accept test suites like the `apex run test` commands does so that `project deploy start` can pass in suites and not hit the request limits

https://github.com/forcedotcom/cli/discussions/2349

**Comment by jshackell-sfdc** [comment]
Released in 2.35.6
